site/source/_posts/2019/01/28/Version-5-Dojo.md

```diff
@@ -0,0 +1,204 @@
+---
+title: Dojo 5
+date: 2019-01-28 08:00:00
+authorId: unknown
+featured_image: featured.jpg
+---
+
+Dojo version 5 has arrived!
+
+We're excited to announce the 5.0.0 release of Dojo. This version builds on previous releases with a vast array of bug fixes and improvements throughout the framework.
+
+{% asset_img featured.jpg feature-full %}
+
+### Conditional Polyfills - Serving Less JavaScript
+
+One of our primary goals for modern Dojo is to improve performance by serving minimal JavaScript bundles by default. For version 5, we're happy to announce an out of the box solution to optimize the user experience for dealing with bundling and loading polyfills in Dojo applications.
+
+By using `@dojo/framework/shim` ponyfills for [`WebAnimations`][WebAnimationsAPI], [`IntersectionObserver`][IntersectionObserverAPI], [`ResizeObserver`][ResizeObserverAPI], and [`fetch`][FetchAPI] the Dojo build produces separate "platform" bundles that will be **only** be loaded based on two key conditions:
+
+1) The shim module is imported somewhere in an application.
+1) A users browser does not natively support the browser feature.
+
+In short, if your application does not use the browser feature or you use a browser with native support, the polyfill will not be loaded! This change means serving less JavaScript and improving your application performance without compromising on features.
+
+### Better Build Time Rendering (BTR)
+
+Build time rendering provides the rendering of your application to HTML during the build and in-lines critical CSS. BTR allows an application to render static HTML pages and offer some advantages of server-side rendering (SSR) such as performance and search engine optimization without the complexities of running a server to support full SSR. Build time rendering has been available via the Dojo [cli-build-app][CliBuildAppReadme]  command since our initial 2.0.0 release.
+
+The 5.0.0 release brings a more robust, flexible and feature-rich build time render by running your application at build time in a browser environment.
+
+Routing support has also been extended to support a `HistoryManager` using the browser [history API][HistoryAPI] via the `@dojo/framework/routing/history/StateHistory`. For `StateHistory` BTR produces a static HTML file for each of the paths defined in the build time rendering configuration.
+
+To configure build time rendering to create static pages for the BTR paths, ensure you are using the `StateHistory` in your application and update your BTR config to specify paths without a `#` prefix.
+
+```json
+"build-app": {
+	"build-time-render": {
+		"root": "app",
+		"paths": [
+			"home",
+			"about",
+		]
+	}
+}
+```
+
+### Introducing Dojo Blocks
+
+Leveraging the improved BTR, Dojo Blocks open the door for creating rich static websites by enabling code to be executed in Node.js at build time and cached in the appropriate bundle for runtime.
```

Comment:
Dojo Blocks is a new mechanism leveraging Build Time Rendering that allows executing code in node.js as part of the build. The results from the execution are written to a cache that can then be transparently used in the same way at runtime in the browser. This opens up new opportunities for performing operations that might be not possible or unperformant in a browser.